src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -434,6 +456,275 @@ def show_ag_identity(cmd, resource_group_name, application_gateway_name):
+
+    # prepare the subnet for new private link
+    for subnet in vnet.subnets:
+        if subnet.name == private_link_subnet_name:
+            raise CLIError('Subnet name duplicates')
+        if subnet.address_prefix == private_link_subnet_prefix:
+            raise CLIError('Subnet prefix duplicates')
+        if subnet.address_prefixes and private_link_subnet_prefix in subnet.address_prefixes:
+            raise CLIError('Subnet prefix duplicates')
+
+    private_link_subnet = Subnet(name=private_link_subnet_name,
+                                 address_prefix=private_link_subnet_prefix,
+                                 private_link_service_network_policies='Disabled')
+    vnet.subnets.append(private_link_subnet)
+    ncf.virtual_networks.create_or_update(resource_group_name, vnet_name, vnet)
+
+    private_link_subnet_id = resource_id(
+        subscription=get_subscription_id(cmd.cli_ctx),
+        resource_group=resource_group_name,
+        namespace='Microsoft.Network',
+        type='virtualNetworks',
+        name=vnet_name,
+        child_type_1='subnets',
+        child_name_1=private_link_subnet_name
+    )
+    private_link_ip_allocation_method = IPAllocationMethod.static.value if private_link_ip_address \
+        else IPAllocationMethod.dynamic.value
+    private_link_ip_config = ApplicationGatewayPrivateLinkIpConfiguration(
+        name='PrivateLinkDefaultIPConfiguration',
+        private_ip_address=private_link_ip_address,
+        private_ip_allocation_method=private_link_ip_allocation_method,
+        subnet=SubResource(id=private_link_subnet_id),
+        primary=private_link_primary
+    )
+    private_link_config = ApplicationGatewayPrivateLinkConfiguration(
+        name=private_link_name,
+        ip_configurations=[private_link_ip_config]
+    )
+
+    # associate the private link with the frontend IP configuration
+    for fic in appgw.frontend_ip_configurations:
+        if fic.name == frontend_ip:
+            fic.private_link_configuration = SubResource(id=private_link_config_id)
+
+    appgw.private_link_configurations.append(private_link_config)
+
+    return ncf.application_gateways.create_or_update(resource_group_name, application_gateway_name, appgw)
+
+
+def show_ag_private_link(cmd,
+                         resource_group_name,
+                         application_gateway_name,
+                         private_link_name):
+    ncf = network_client_factory(cmd.cli_ctx)
+
+    appgw = ncf.application_gateways.get(resource_group_name, application_gateway_name)
+
+    target_private_link = None
+    for pl in appgw.private_link_configurations:
+        if pl.name == private_link_name:
+            target_private_link = pl
+            break
+    else:
+        raise CLIError("Priavte Link doesn't exist")
+
+    return target_private_link
+
+
+def list_ag_private_link(cmd,
+                         resource_group_name,
+                         application_gateway_name):
```

Comment:
Please set this one with `id_part=None` under `list` command

Reply:
fixed